lib/matplotlib/axes/_base.py

```diff
@@ -945,6 +946,42 @@ def _gen_axes_spines(self, locations=None, offset=0.0, units='inches'):
         return OrderedDict((side, mspines.Spine.linear_spine(self, side))
                            for side in ['left', 'right', 'bottom', 'top'])
 
+    def sharex(self, other):
+        """
+        Share the x-axis with *other*.
+
+        This is equivalent to passing ``sharex=other`` when constructing the
```

Comment:
This docstring could be more explicit and non circular.  For instance the verb to share would indicate that self shares with other, whereas the exact opposite happens.  Further passing sharex=other calls this function so I don’t think it’s very helpful to say that this function is the same as that and expect the user to ferret out the behaviour

Reply:
"self shares with other, whereas the exact opposite happens"
What do you mean by "opposite"?  Isn't sharing symmetrical? (well perhaps not deep in the implementation details, but from the user PoV it is)

Reply:
I meant in everyday English the transitive verb “to share” means one person shares Something with another

Reply:
Can you propose an alternate wording here?

Reply:
Sorry, the main point is that saying `ax.sharex` is "Equivalent to passing sharex=other when constructing..." is a circular doc, because when you construct, you call `ax.sharex`.  This function ideally would define what `ax.sharex` does explicitly.

Reply:
Right now the docs of sharex (as kwargs to the Axes constructor) are
```
        sharex, sharey : `~.axes.Axes`, optional
            The x or y `~.matplotlib.axis` is shared with the x or
            y axis in the input `~.axes.Axes`.
```
which is not much better :/ I agree this can be improved, but is arguably a separate issue?

Reply:
The docstring defines what the function does for the user, and one of our main problems with "share" is that people don't know what it means and even in the history of the code base different definitions have popped up.  Right now I'm not 100% sure what this PR changed and I think it should be defined clearly, in plain English, what calling this function will do.

Reply:
```
ax0 = (some axes)
ax1 = fig.add_subplot(...)
ax1.sharex(ax0)
```
is equivalent to
```
ax0 = (some axes)
ax1 = fig.add_subplot(..., sharex=ax0)
```
Now it is true that what the latter does is not really well documented either, but the doc available for `ax1.sharex(ax0)` is, well, exactly the same.